File: 1063-best-sightseeing-pair/best-sightseeing-pair.py
class Solution:
    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        LeftMax = values[0]
        ans = 0

        for i in range (1,len(values)):
            ans = max(ans, LeftMax+values[i]-i)

            LeftMax = max(LeftMax, values[i]+i)

        return ans

File: 1063-best-sightseeing-pair/best-sightseeing-pair.cpp
class Solution {
public:
    int maxScoreSightseeingPair(vector<int>& values) {
        int LeftMax = values[0];
        int ans = 0;

        for(int i=1; i<values.size(); i++){
            ans = max(ans, LeftMax + values[i] - i);
            
            LeftMax = max(LeftMax, values[i] + i);
        }

        return ans;
    }
};